esx_service/vmdk_ops.py

```diff
@@ -634,9 +645,10 @@ def disk_detach(vmdk_path, vm):
     dev_changes.append(disk_spec)
     spec.deviceChange = dev_changes
 
+    # Handle the most generic exception for a VM reconfig.
     try:
         wait_for_tasks(si, [vm.ReconfigVM_Task(spec=spec)])
-    except vim.fault.GenericVmConfigFault as ex:
+    except vim.fault.VimFault as ex:
```

Comment:
It seems consistent to change GenericVmConfigFault =>VimFault. However, this doesn't really fix #381.

Unrelated to your change but does `logging.warning(f.message)` logs exception code too?